From f438777263673d8344f99383ef0f13dbe482c6ea Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Wed, 3 Jan 2018 19:38:06 +0200
Subject: [PATCH 2151/2241] fix: phy: marvell: fix incorrect comphy 4 selector
 setting

Currently, comphy 4 has incorrect setting for SGMII mode.
This patch sets the correct setting according to the
reference manual:
	COMPHY4 PHY Protocol Selector
		0 = Unconnected
		1 = SGMII/HS-SGMII Port2
		2 = SGMII/HS-SGMII Port1: XFI/10G, RXAUI_Lane0

No changes for XFI/RXAUI modes.

Change-Id: Ie011aeb9139d0aaac5bf8071ffaf854251afa4d8
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/48505
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/phy/phy-comphy-cp110.c | 12 ++++++------
 drivers/phy/phy-comphy-cp110.h |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/phy/phy-comphy-cp110.c b/drivers/phy/phy-comphy-cp110.c
index 423778e..70bb209 100644
--- a/drivers/phy/phy-comphy-cp110.c
+++ b/drivers/phy/phy-comphy-cp110.c
@@ -99,8 +99,8 @@ static void mvebu_cp110_comphy_set_phy_selector(struct mvebu_comphy_priv *priv,
 			break;
 		case(4):
 			 /* For comphy 4:
-			  * 0x1 = SGMII/HS-SGMII Port1
-			  * 0x2 = SGMII/HS-SGMII Port0: XFI/SFI, RXAUI_Lane0
+			  * 0x1 = SGMII/HS-SGMII Port2
+			  * 0x2 = SGMII/HS-SGMII Port1: XFI/SFI, RXAUI_Lane0
 			  *
 			  * We want to check if SGMII1/HS_SGMII1 is the requested mode in order to
 			  * determine which value should be set (all other modes use the same value)
@@ -108,8 +108,8 @@ static void mvebu_cp110_comphy_set_phy_selector(struct mvebu_comphy_priv *priv,
 			  * SGMII0/HS_SGMII0 too.
 			  */
 			if ((mode == COMPHY_SGMII_MODE || mode == COMPHY_HS_SGMII_MODE) &&
-			    COMPHY_GET_ID(priv->lanes[comphy->index].mode) == 1)
-				reg |= COMMON_SELECTOR_COMPHY4_SGMII1 << comphy_offset;
+			    COMPHY_GET_ID(priv->lanes[comphy->index].mode) == 2)
+				reg |= COMMON_SELECTOR_COMPHY4_SGMII2 << comphy_offset;
 			else
 				reg |= COMMON_SELECTOR_COMPHY4_ALL_OTHERS << comphy_offset;
 			break;
@@ -2064,8 +2064,8 @@ const struct mvebu_comphy_soc_info cp110_comphy = {
 		{COMPHY_UNUSED, COMPHY_SGMII1, COMPHY_HS_SGMII1, COMPHY_RXAUI1,
 		 COMPHY_SATA1, COMPHY_USB3H1, COMPHY_PCIE0},
 		/* Lane 4 */
-		{COMPHY_UNUSED, COMPHY_SGMII0, COMPHY_HS_SGMII0, COMPHY_SGMII1,
-		 COMPHY_HS_SGMII1, COMPHY_RXAUI0, COMPHY_XFI, COMPHY_SFI,
+		{COMPHY_UNUSED, COMPHY_SGMII1, COMPHY_HS_SGMII1, COMPHY_SGMII2,
+		 COMPHY_HS_SGMII2, COMPHY_RXAUI0, COMPHY_XFI, COMPHY_SFI,
 		 COMPHY_USB3H1, COMPHY_USB3D0, COMPHY_PCIE1},
 		/* Lane 5 */
 		{COMPHY_UNUSED, COMPHY_RXAUI1, COMPHY_SGMII2, COMPHY_HS_SGMII2,
diff --git a/drivers/phy/phy-comphy-cp110.h b/drivers/phy/phy-comphy-cp110.h
index a6d380a..9b92e6f 100644
--- a/drivers/phy/phy-comphy-cp110.h
+++ b/drivers/phy/phy-comphy-cp110.h
@@ -44,7 +44,7 @@ extern const struct mvebu_comphy_soc_info cp110_comphy;
 #define COMMON_SELECTOR_COMPHY0_1_2_NETWORK	0x1
 #define COMMON_SELECTOR_COMPHY3_RXAUI		0x1
 #define COMMON_SELECTOR_COMPHY3_SGMII		0x2
-#define COMMON_SELECTOR_COMPHY4_SGMII1		0x1
+#define COMMON_SELECTOR_COMPHY4_SGMII2		0x1
 #define COMMON_SELECTOR_COMPHY4_ALL_OTHERS	0x2
 #define COMMON_SELECTOR_COMPHY5_RXAUI		0x2
 #define COMMON_SELECTOR_COMPHY5_SGMII		0x1
-- 
2.7.4

